android/quest/src/main/java/org/smartregister/fhircore/quest/QuestApplication.kt

```diff
@@ -75,4 +83,28 @@ class QuestApplication : Application(), DataCaptureConfig.Provider, Configuratio
       .setMinimumLoggingLevel(if (BuildConfig.DEBUG) Log.VERBOSE else Log.INFO)
       .setWorkerFactory(workerFactory)
       .build()
+
+  private val globalExceptionHandler =
+    Thread.UncaughtExceptionHandler { _: Thread, e: Throwable -> handleUncaughtException(e) }
+
+  /**
+   * This method captures all uncaught exceptions in the app and redirects to the Launch Page in the
+   * case that the exception was thrown on the main thread This will therefore prevent any app
+   * crashes so we need some more handling for reporting the errors once we have a crash manager
+   * installed
+   *
+   * TODO add crash reporting when a crash reporting tool is selected e.g. Fabric Crashlytics or
+   * Sentry
+   */
+  private fun handleUncaughtException(e: Throwable) {
+    showToast(this.getString(R.string.error_occurred))
+    Timber.e(e)
+
+    if (Looper.myLooper() == Looper.getMainLooper()) {
+      val intent = Intent(applicationContext, AppSettingActivity::class.java)
```

Comment:
My assumption is this would then redirect the user to the PIN or username/password login page since the App Ids were already entered into the app. If yes I think that's fine but we can consult the Design team. 

I think it would be better to redirect to the home page assuming the crash is happening somewhere else other that the home page. We could redirect to the Login pages if the crashes happen on the home page. We could then redirect to the setting page if they happen on the login pages.

My reasoning for the above proposal is to still give the use a change to use the app if some pieces of it are usable. 

I realise it might be too much or over engineering of some sought though.

Reply:
agreed, we should consider options here, moving to this issue, https://github.com/opensrp/fhircore/issues/2193